src/terrain/index.js

```diff
@@ -0,0 +1,157 @@
+const { getRandomTerrainFunction } = require('./terrainFunctions');
+const terrainGenerator = (boardBounds = { x: 100, y: 100 }, terrainBounds = { x: 5, y: 5 }, numberOfTerrainElements = 5) => {
+    let { mockArray, mockWidth, mockHeight } = createMockArray(boardBounds, terrainBounds);
+    const keyPoint = {
+        x: Math.floor(Math.random() * mockWidth),
+        y: Math.floor(Math.random() * mockHeight)
+    };
+    mockArray = generateTerrain(mockArray, keyPoint, numberOfTerrainElements, mockWidth, mockHeight);
+    return mockArray;
+};
+
+createMockArray = (boardBounds, terrainBounds) => {
+    const boardX = boardBounds.x;
+    const boardY = boardBounds.y;
+    const terrainX = terrainBounds.x;
+    const terrainY = terrainBounds.y;
+
+    const mockArray = [];
+    const mockWidth =  boardX/terrainX;
+    const mockHeight = boardY/terrainY;
+
+    const defaultInfo = {
+        filled: false,
+        north: true,
+        south: true,
+        east: true,
+        west: true
+    };
+
+    for(let i = 0; i < mockWidth; i++) {
+        mockArray[i] = [];
+        for (let j = 0; j < mockHeight; j++) {
+            const info = { ...defaultInfo };
+            if (i === 0) {
+                info.west = false;
+            } else if (i === mockWidth - 1) {
+                info.east = false;
+            }
+            if (j === 0) {
+                info.north = false;
+            } else if (j === mockHeight - 1) {
+                info.south = false;
+            }
+            mockArray[i].push(info);
+        }
+    }
+    return {
+        mockArray,
+        mockWidth,
+        mockHeight
+    };
+};
+
+const deepClone = (toCopy) => {
```

Comment:
If I understand correctly, this is just creating a new array of objects, but isn't necessarily a "deep clone" since it's not cloning those objects

Reply:
this actually is a deep clone of the board with the way the combo of spread and .map works 

see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals